src/codewhisperer/activation.ts

```diff
@@ -98,14 +102,17 @@ export async function activate(context: ExtContext): Promise<void> {
                     await set(CodeWhispererConstants.termsAcceptedKey, false, context.extensionContext.globalState)
                     await set(CodeWhispererConstants.autoTriggerEnabledKey, false, context.extensionContext.globalState)
                     if (!isCloud9()) {
-                        InlineCompletion.instance.hideCodeWhispererStatusBar()
+                        hideStatusBar()
                     }
                 }
                 vscode.commands.executeCommand('aws.codeWhisperer.refresh')
             }
             if (configurationChangeEvent.affectsConfiguration('aws.codeWhisperer')) {
                 referenceLogViewProvider.update()
             }
+            if (configurationChangeEvent.affectsConfiguration('editor.inlineSuggest.enabled')) {
+                await vscode.commands.executeCommand('workbench.action.reloadWindow')
+            }
```

Comment:
We have to ask the user before reloading the window automatically. Could show a notification, something like: "Inline suggestion settings changed. The current window needs to be reloaded for this to take affect." with a button "Reload Now".

Reply:
Added this button in https://github.com/aws/aws-toolkit-vscode/pull/2888/commits/bd730768b2a7e60147a88ecfb7974f103b0e8c32